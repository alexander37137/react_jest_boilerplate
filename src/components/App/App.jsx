import React, { Component } from 'react';
import {
  Tab,
  TabList,
  TabPanel,
  Tabs,
} from 'react-tabs';
import 'react-tabs/style/react-tabs.css';

import './App.css';

// eslint-disable-next-line react/prefer-stateless-function
class App extends Component {
  render() {
    return (
      <Tabs>
        <TabList>
          <Tab>
            <span data-artists-tab>Artists</span>
          </Tab>
          <Tab>
            <span data-books-tab>Books</span>
          </Tab>
        </TabList>

        <TabPanel>
          <h2 data-tab-content>Any content 1</h2>
        </TabPanel>
        <TabPanel>
          <h2 data-tab-content>Any content 2</h2>
        </TabPanel>
      </Tabs>
    );
  }
}

export default App;
