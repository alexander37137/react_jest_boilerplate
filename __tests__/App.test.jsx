import React from 'react';
import { mount } from 'enzyme';

import App from '../src/components/App';

describe('<App />', () => {
  it('renders app', () => {
    const wrapper = mount(<App />);

    const tabContent = wrapper.find('[data-tab-content]');
    expect(tabContent).toMatchSnapshot();
  });

  it('selects second tab', () => {
    const wrapper = mount(<App />);
    const tab = wrapper.find('[data-books-tab]');
    tab.simulate('click');
    const tabContent = wrapper.find('[data-tab-content]');
    expect(tabContent).toMatchSnapshot();
  });
});
