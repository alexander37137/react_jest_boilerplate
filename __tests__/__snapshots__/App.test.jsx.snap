// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders app 1`] = `
<h2
  data-tab-content={true}
>
  Any content 1
</h2>
`;

exports[`<App /> selects second tab 1`] = `
<h2
  data-tab-content={true}
>
  Any content 2
</h2>
`;
